sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -54,52 +90,72 @@ import org.apache.spark.unsafe.types.UTF8String
   """,
   since = "3.3.0",
   group = "string_funcs")
-case class ToNumber(left: Expression, right: Expression)
-  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
-
-  private lazy val numberFormat = right.eval().toString.toUpperCase(Locale.ROOT)
-  private lazy val numberFormatter = new NumberFormatter(numberFormat)
+case class ToNumber(left: Expression, right: Expression) extends NumberFormatterBase {
+  val isParse: Boolean = true
```

Comment:
I am a bit favor use a function to get the method name in codegen than relying on `isParse`. `isParse` can only be yes or no but say we could have a third implementation extends `NumberFormatterBase`. At that time `isParse` won't be sufficient.

Also `isParse` is also a bit vague.